Demo/Common/IntegrationExampleComponents.swift

```diff
@@ -128,8 +128,12 @@ extension IntegrationExample {
         }
     }
 
-    private func handle(_ action: Action) {
-        actionComponent.handle(action)
+    internal func handle(_ action: Action) {
+        if let actionComponent = currentComponent as? ActionHandlingComponent {
+            actionComponent.handle(action)
+        } else {
+            actionComponent.handle(action)
+        }
```

Comment:
I have left my "5 reasons we should not do this" in previous PR 😆

Reply:
what's the alternative? :-), the old way simply doesn't work, I replied to your comment in the old PR 😁

Reply:
hm.. ok, I see

We can rename `internal lazy var actionComponent: AdyenActionComponent` into `internal lazy var adyenActionComponent: AdyenActionComponent`

OR change local `let` name

```suggestion
        if let dropInAsActionComponent = currentComponent as? ActionHandlingComponent {
            dropInAsActionComponent.handle(action)
        } else {
            actionComponent.handle(action)
        }
```

OR (as we discussed yesterday)

move each integration (DropIn, Component, Session) in a standalone class with it's own set of local variables